### PR TITLE
Display time in transaction history list

### LINKED ISSUE #2131: Wallet - add timestamp and recipient name to the transaction history
Per updated mockup
https://zpl.io/25ELQ8j

### PR DESCRIPTION
**Pull request by janherich**
Also make `[list/item-content]` element more modular and reusable
as it was necessary to accomplish the task.

fixes #2131 

### Steps to test:
- Check transaction history screen (also unsigned transactions screen)

status: ready

### CODE REVIEW

File: src/status_im/ui/screens/wallet/transactions/styles.cljs [hunk 1]
```diff
@@ -51,6 +51,20 @@
    :margin-top        22
    :margin-horizontal 92})
 
+(def amount-time
```
Comment (flexsurfer) [inline src/status_im/ui/screens/wallet/transactions/styles.cljs]: please use `defstyles` macro

**Review by flexsurfer — CHANGES_REQUESTED** [review]

File: src/status_im/ui/screens/wallet/transactions/styles.cljs [hunk 2]
```diff
@@ -51,6 +51,20 @@
    :margin-top        22
    :margin-horizontal 92})
 
+(defstyle amount-time
```
Comment (flexsurfer) [inline src/status_im/ui/screens/wallet/transactions/styles.cljs:54]: `defstyle` is used for platform differences, 
```
(defstyle amount-time
 {:flex-direction  :row
  :justify-content :space-between
  :ios {:padding-top 13}
  :android {:padding-top 14}})
```
you don't need it for tx-amount and tx-time

**Comment by flexsurfer** [comment]
@janherich please make changes for `amount-time` and squash commits

**Comment by janherich** [comment]
@flexsurfer done

**Review by flexsurfer — APPROVED** [review]

**Review by jeluard — APPROVED** [review]

**Comment by annadanchenko** [comment]
There are 3 issues:
1. on iOS and Android with 24-hour format (so it shows 19:34 for evening) I see transaction time shown as 07:34. Would expect that it will be shown or in 24 hour format (preferably as it’s shown in such format on Details screen) or that am/pm will be added to the time.

Transactions done in the evening (19:32 and 19:35) but shown as 07:32 and 07:35:
img_2360

2. Currently,  if sum is too long then amount and time are very close to each other. Need to add ... to the end of the sum if it can't fit the line so there is notification for user that it's incomplete sum and there is enough space between amount and time. See https://zpl.io/25ELQ8j 
Picture with current state (issue):
img_2365

3. Username is grey (at least on iOS), should be black

**Comment by annadanchenko** [comment]
next build: 
changed time to 20:11 on my iOS device before installing the build (to see 24 hour format) - on Unsigned it shows 20:11 but on History it’s still 11:11. Would expect for both lists to show the time set on device. TF session https://app.testfairy.com/projects/4803590-status/builds/6879469/sessions/2/?accessToken=td2R8L9zp2A7XJRftgRZApkrUOU

**Comment by janherich** [comment]
@annadanchenko All the remaining issues are solved in the branch

File: src/status_im/components/list/views.cljs [hunk 3]
```diff
@@ -54,15 +56,21 @@
...
+  [primary]
+  [rn/text {:style lst/primary-text} primary])
+
+(defn item-primary-only
+  [primary]
+  [rn/text {:style lst/primary-text-only} primary])
+
+(defn item-secondary
+  [secondary]
+  [rn/text {:style lst/secondary-text :ellipsize-mode "middle" :number-of-lines 1} secondary])
```
Comment (rasom) [inline src/status_im/components/list/views.cljs:69]: not a big deal but we could use `:middle` here

File: src/status_im/ui/screens/wallet/transactions/styles.cljs [hunk 4]
```diff
@@ -51,6 +51,45 @@
...
+  {:font-size 16
+   :color     styles/color-gray4})
+
+(def address-label
+  (merge address-item
+         {:margin-right 5}))
+
+(def address-contact
+  (merge address-item
+         {:color styles/color-black
```
Comment (rasom) [inline src/status_im/ui/screens/wallet/transactions/styles.cljs]: alignment

File: src/status_im/ui/screens/wallet/transactions/views.cljs [hunk 5]
```diff
@@ -76,21 +76,37 @@
...
+  (let [[label contact address] (if (inbound? type)
+                                  [(i18n/label :t/from) from-contact from]
+                                  [(i18n/label :t/to) to-contact to])]
+    [list/touchable-item #(re-frame/dispatch [:show-transaction-details hash])
+     [react/view
+      [list/item
+       [list/item-icon (transaction-type->icon (keyword type))]
+       [list/item-content
+        [react/view {:style transactions.styles/amount-time}
+         [react/text {:style transactions.styles/tx-amount
```
Comment (rasom) [inline src/status_im/ui/screens/wallet/transactions/views.cljs:84]: alignment in map

**Comment by annadanchenko** [comment]
long amount is shown with … in the middle but it’s expected to get “…” at the end of a sum, e.g. : `1234567.09876... ETH`. Now it’s `1234...876 ETH`

**Pull request merged**